--- z88dk-1.7.ds1/libsrc/Makefile.orig	2007-09-25 16:39:30.000000000 +0200
+++ z88dk-1.7.ds1/libsrc/Makefile	2007-09-25 16:39:43.000000000 +0200
@@ -20,7 +20,7 @@
 	ozansi_clib.lib rex_clib.lib rcmx000_clib.lib \
 	svi_clib.lib sam_clib.lib samansi_clib.lib \
 	pps_clib.lib ppsansi_clib.lib embedded_clib.lib \
-	zx81_clib.lib zx81ansi_clib.lib
+	zx81_clib.lib zx81ansi_clib.lib sms_clib.lib
 
 genlibs: gen_math.lib malloc.lib z80_crt0.lib ndos.lib adt.lib balloc.lib \
          im2.lib rect.lib algorithm.lib
