--- z88dk-1.7.ds1/libsrc/zxmath/Makefile.orig	2007-06-03 17:13:07.000000000 +0200
+++ z88dk-1.7.ds1/libsrc/zxmath/Makefile	2008-02-27 11:22:42.000000000 +0100
@@ -9,19 +9,19 @@
 
 zx_mathtiny:
 	rm -f *.o* ../mzx_tiny.lib
-	$(LIBLINKER) -DFORzx -DTINYMODE -x$(OUTPUT_DIRECTORY)/mzx_tiny @zxmlist
+	$(LIBLINKER) -DFORzx -DTINYMODE -x../$(OUTPUT_DIRECTORY)/mzx_tiny @zxmlist
 
 zx81_mathtiny:
 	rm -f *.o* ../m81_tiny.lib
-	$(LIBLINKER) -DFORzx81 -DTINYMODE -x$(OUTPUT_DIRECTORY)/m81_tiny @zxmlist
+	$(LIBLINKER) -DFORzx81 -DTINYMODE -x../$(OUTPUT_DIRECTORY)/m81_tiny @zxmlist
 
 zx_math:
 	rm -f *.o* ../mzx.lib
-	$(LIBLINKER) -DFORzx -x$(OUTPUT_DIRECTORY)/mzx @zxmlist
+	$(LIBLINKER) -DFORzx -x../$(OUTPUT_DIRECTORY)/mzx @zxmlist
 
 zx81_math:
 	rm -f *.o* ../m81.lib
-	$(LIBLINKER) -DFORzx81 -x$(OUTPUT_DIRECTORY)/m81 @zxmlist
+	$(LIBLINKER) -DFORzx81 -x../$(OUTPUT_DIRECTORY)/m81 @zxmlist
 
 clean:
 	rm -f *.o* *.sym *.map *.err 
