--- z88dk-1.6.ds1/src/sccz80/stmt.c.orig
+++ z88dk-1.6.ds1/src/sccz80/stmt.c
@@ -378,10 +378,10 @@
 
 		jump(l_condition);		/*         goto condition             */
 		postlabel(wq.loop);		/* .loop                              */
-		statement();			/*         statement                  */
 		clearbuffer(buf3);		/*         modification               */
-		postlabel(l_condition);	/* .condition                         */
+		postlabel(l_condition);		/* .condition                         */
 		clearbuffer(buf2);		/*         if (!condition) goto exit  */
+		statement();			/*         statement                  */
 		jump(wq.loop);			/*         goto loop                  */
 		postlabel(wq.exit);		/* .exit                              */
 
